src/data_hub/api.py

```diff
@@ -861,14 +861,22 @@ def get_dataframe(
 
         # this import is here because we don't want to
         # import pandas unless we actually use this function
-        import pandas as pd
+        from deeporigin.data_hub.dataframe import DataFrame
 
-        df = pd.DataFrame(data)
+        df = DataFrame(data)
         df.attrs["file_ids"] = list(set(file_ids))
         df.attrs["reference_ids"] = list(set(reference_ids))
         df.attrs["id"] = database_id
+        df.attrs["metadata"] = dict(db_row)
 
-        return _type_and_cleanup_dataframe(df, columns)
+        df = _type_and_cleanup_dataframe(df, columns)
+
+        # if this code is running the lambda, we do not
+        # turn on auto_sync by default because we don't want
+        # side effects from code that the LLM writes
+        if os.environ.get("AWS_LAMBDA_FUNCTION_NAME") is None:
+            df.auto_sync = True
+        return df
```

Comment:
really don't want to allow the LLM to accidentally modify DBs by doing something it thinks is local

Reply:
Conversely, it could be quite convenient for a user to write this code in a code executor block and expect it to work. But I do agree that this is super dangerous in general. Perhaps let's start with this safe implementation and strategize how to deal with the possibility of writes.

Reply:
they can get the "automatic" functionality to work using the `auto_sync` attribute

Reply:
> I do agree that this is super dangerous in general

Highly agree. I love the spirit of the PR. With a couple minor changes (see suggestions [below](https://github.com/deeporiginbio/deeporigin-client/pull/72#pullrequestreview-2244151789)), I think we can provide a UX which is (a) safe, (b) simpler, and (c) highly consistent with the way that DataFrames are frequently used.